src/ui/components/pages/assets/tabs/tabAssets.tsx

```diff
@@ -0,0 +1,121 @@
+import cn from 'classnames';
+import * as styles from '../../styles/assets.styl';
+import { colors, icontains } from '../helpers';
+import { Trans } from 'react-i18next';
+import { AssetItem } from '../assetItem';
+import { Asset, Money } from '@waves/data-entities';
+import { BigNumber } from '@waves/bignumber';
+import * as React from 'react';
+import { SearchInput } from '../../Assets';
+import { useAppSelector } from '../../../../store';
+import { TabPanel } from '../../../ui';
+import { useAssetFilter } from './helpers';
+
+export function TabAssets({ onItemClick }) {
+  const assets = useAppSelector(state => state.assets);
+  const address = useAppSelector(state => state.selectedAccount.address);
+  const myAssets = useAppSelector(
+    state => state.balances[address]?.assets || {}
+  );
+  const showSuspiciousAssets = useAppSelector(
+    state => !!state.uiState?.showSuspiciousAssets
+  );
+
+  const [assetTerm, setAssetTerm] = useAssetFilter('term');
+  const [onlyMyAssets, setOnlyMyAssets] = useAssetFilter('onlyMy');
+  const [onlyFavorites, setOnlyFavorites] = useAssetFilter('onlyFavorites');
+
+  const assetEntries = Object.entries<{ balance: string }>(myAssets)
+    .filter(
+      ([assetId]) =>
+        (showSuspiciousAssets || !assets[assetId]?.isSuspicious) &&
+        (!onlyFavorites || assets[assetId]?.isFavorite === onlyFavorites) &&
+        (!onlyMyAssets || assets[assetId]?.issuer === address) &&
+        (!assetTerm ||
+          assetId === assetTerm ||
+          icontains(assets[assetId]?.displayName, assetTerm))
+    )
+    .sort(
+      ([a], [b]) =>
+        (assets[a] &&
+          assets[b] &&
+          +!!assets[b].isFavorite - +!!assets[a].isFavorite) ||
```

Comment:
I guess parenthesis are grouped incorrectly here. Opening paren should be on this line, not before the first condition, where you check for assets presence. Otherwise, if there's no assets for these ids, the following conditions (e.g. `isSuspicious`) are checked, reading props from `undefined`. I got an error when I open the popup for the first time after this change, before assets are loaded.

Reply:
Sure, already fixed this